jax/_src/ad_checkpoint.py

```diff
@@ -124,10 +124,10 @@ def save_and_offload_only_these_names(
   names_which_can_be_saved = set(names_which_can_be_saved)
   names_which_can_be_offloaded = set(names_which_can_be_offloaded)
   def policy(prim, *_, **params):
-    if prim is name_p and params['name'] in names_which_can_be_saved:
-      return pe.Saveable
     if prim is name_p and params['name'] in names_which_can_be_offloaded:
       return pe.Offloadable(src=offload_src, dst=offload_dst)
+    if prim is name_p and params['name'] in names_which_can_be_saved:
```

Comment:
Maybe instead, we can check and raise an error if names appear in both of them?

Reply:
But why raise an error when offloading implicitly assumes checkpointing so from user's perspective it should be WAI?

Reply:
Not really? Here save means save to HBM and offload is to offload to host DRAM. 

So if you do say `names_which_can_be_saved=["x"]` and the same name appears in offload, what does that mean? It's ambiguous right?

Reply:
I think it should be reworded then to be "names_which_can_be_saved_to_hbm".

Reply:
That sounds good to me. I wonder if @mattjj has any opinions here.

Reply:
Discussed offline, will resubmit with a seprate policy.